pkg/domain/attributor.go

```diff
@@ -42,6 +44,57 @@ type NexposeAttributedAssetVulnerabilities struct {
 	BusinessContext CloudAssetDetails `json:"businessContext"`
 }
 
+// UnmarshalJSON is a custom unmarshaller to ensure no `nil` or `null` fields in the resulting struct
+func (n *NexposeAttributedAssetVulnerabilities) UnmarshalJSON(data []byte) error {
+	type Alias NexposeAttributedAssetVulnerabilities
+	aux := &struct {
+		*Alias
+	}{
+		Alias: (*Alias)(n),
+	}
```

Comment:
Q: Wait, what just happened?

Reply:
Yes, please.

Reply:
I have to use a type alias because I'm actually calling `json.Unmarshal` _inside_ the `UnmarshalJSON` function (see a few lines down).  That risks an infinite recursion because the target memory address in that `json.Unmarshal` function call _is the struct to which this custom unmarshaller function is attached_.

The `Alias` therefore is a very temporary type that does _not_ have the custom `UnmarshalJSON` function attached (because type aliases do not inherit attached functions), thus avoiding the infinite recursion.